synapse/handlers/federation.py

```diff
@@ -226,15 +226,31 @@ async def _maybe_backfill_inner(
         """
         backwards_extremities = [
             _BackfillPoint(event_id, depth, _BackfillPointType.BACKWARDS_EXTREMITY)
-            for event_id, depth in await self.store.get_backfill_points_in_room(room_id)
+            for event_id, depth in await self.store.get_backfill_points_in_room(
+                room_id=room_id,
+                current_depth=current_depth,
+                # We only need to end up with 5 extremities combined with the
+                # insertion event extremities to make the `/backfill` request
+                # but fetch an order of magnitude more to make sure there is
+                # enough even after we filter them by whether visible in the
+                # history. This isn't fool-proof as all backfill points within
+                # our limit could be filtered out but seems like a good amount
+                # to try with at least.
+                limit=50,
```

Comment:
if `limit` is always 50... why not hardcode it in `get_backfill_points_in_room` ?

Reply:
It seems more obvious this way. It's not obvious asking `get_backfill_points_in_room`  and it only returns the 50 most relevant events if that was a default.